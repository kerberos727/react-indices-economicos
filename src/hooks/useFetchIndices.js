import { useEffect, useState } from "react";
import { getIndices } from "../services/api";

export const useFetchIndices = () =>{

  const [indices, setIndices] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const getData = async () => {
    const data = await getIndices();
    setIndices(data);
    setIsLoading(false);
  };

  useEffect(() => {
    getData();
  }, []);

    return {
        indices,
        isLoading
    }
}