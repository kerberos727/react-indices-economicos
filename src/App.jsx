import { ItemGrid } from "./components/ItemGrid";
import { useFetchIndices } from "./hooks/useFetchIndices";

const App = () => {
  const { indices, isLoading } = useFetchIndices();

  return (
    <>
      <h1>Indices Economicos Chilenos</h1>
      <hr />
      {isLoading && <h1>Cargando...</h1>}
      <div className="card-grid">
        {indices.map((i) => (
          <ItemGrid key={i.codigo} indice={i} />
        ))}
      </div>
    </>
  );
};

export default App;
