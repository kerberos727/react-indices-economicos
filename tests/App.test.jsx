import { render, screen } from "@testing-library/react";
import App from "../src/App";
import { useFetchIndices } from "../src/hooks/useFetchIndices";

jest.mock("../src/hooks/useFetchIndices");

describe("App Component", () => {
  test("should show loading... to init", () => {
    useFetchIndices.mockReturnValue({
      indices: [],
      isLoading: true,
    });
    render(<App />);
    expect(screen.getByText("Cargando..."));
  });
  test("should show data when load data finished", () => {
    const mockIndices = [
      {
        codigo: "uf",
        nombre: "Unidad de fomento (UF)",
        unidad_medida: "Pesos",
        fecha: "2022-08-21T04:00:00.000Z",
        valor: 33685.1,
      },
      {
        codigo: "ivp",
        nombre: "Indice de valor promedio (IVP)",
        unidad_medida: "Pesos",
        fecha: "2022-08-21T04:00:00.000Z",
        valor: 34223.11,
      },
    ];
    useFetchIndices.mockReturnValue({
      indices: mockIndices,
      isLoading: true,
    });
    render(<App />);
    expect(screen.getAllByTestId("title").length).toBe(2);
  });

  it("match with snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
