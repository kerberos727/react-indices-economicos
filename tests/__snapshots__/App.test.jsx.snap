// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component match with snapshot 1`] = `
<div>
  <h1>
    Indices Economicos Chilenos
  </h1>
  <hr />
  <h1>
    Cargando...
  </h1>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <div
        class="title"
        data-testid="title"
      >
        Unidad de fomento (UF)
      </div>
      <div
        class="value"
      >
        33,685.1
      </div>
      <div
        class="detail"
      >
        Pesos
        <br />
        Sun Aug 21 2022
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="title"
        data-testid="title"
      >
        Indice de valor promedio (IVP)
      </div>
      <div
        class="value"
      >
        34,223.11
      </div>
      <div
        class="detail"
      >
        Pesos
        <br />
        Sun Aug 21 2022
      </div>
    </div>
  </div>
</div>
`;
